orderbook/orderbook.go

```diff
@@ -0,0 +1,140 @@
+package orderbook
+
+import (
+	"sort"
+	"sync"
+
+	gdax "github.com/banaio/go-gdax"
+)
+
+// Orderbook represents the state of the orderbook.
+type Orderbook struct {
+	bids  map[float64]float64
```

Comment:
Just curious, you are using a float type as the map key. Would this cause problem due to precision? For example, a L2 update message doesn't correctly remove/update an existing entry due to a tiny difference in the price.